src/Testcontainers.Pulsar/PulsarContainer.cs

```diff
@@ -58,9 +58,14 @@ public async Task<string> CreateAuthenticationTokenAsync(TimeSpan expire = defau
             "--subject",
             PulsarBuilder.Username,
             "--expiry-time",
-            $"{expire.TotalSeconds}s",
+            "",
         };
 
+        if (_configuration.Image.FullName is "apachepulsar/pulsar:3.2.0" or "apachepulsar/pulsar:3.2.1" or "apachepulsar/pulsar:3.2.2" or "apachepulsar/pulsar:3.2.3")
+            command[8] = $"{expire.TotalSeconds * 1000}s";
+        else
+            command[8] = $"{expire.TotalSeconds}s";
+
```

Comment:
QQ: Cannot we simply use something like: `$"{expire.TotalMilliseconds}ms"` for all versions?

Reply:
My assumption is that if the flag `--expiry-time` supports `y` and `s`, does it also support `ms`? Which units of time are supported? Does the mentioned bug only occur for `s`? What about days — can we use something like `0.0416d`? I am not very happy with the current workaround because a new Pulsar version without "the fix" will fail again.

Reply:
> My assumption is that if the flag `--expiry-time` supports `y` and `s`, does it also support `ms`?

As you can see in the [source code](https://github.com/apache/pulsar/blob/c39f9f82b425c66c899f818583714c9c98d3e213/pulsar-cli-utils/src/main/java/org/apache/pulsar/cli/converters/RelativeTimeUtil.java), milliseconds (ms) are currently not supported. I’m actively working on a pull request to address this issue. Once fixed, the lowest supported unit will remain seconds (s). Because it's a fix for a regression.

As you can see ms is not supported. When it gets fixed in pulsar ( I am working on a PR ), the lowest unit is s for second.

> Which units of time are supported?

If you examine the referenced Java file, you’ll find that the following time units are supported: seconds (s), minutes (m), hours (h), days (d), weeks (w), and years (y).

> Does the mentioned bug only occur for `s`? What about days — can we use something like `0.0416d` (~ 1 hour)?

The mentioned bug affects all time units, not just seconds

> I am not very happy with the current workaround because a new Pulsar version without "the fix" will fail again.

I agree that the current workaround isn’t ideal because a new Pulsar version without the fix will encounter the same issue. To mitigate this, we could consider setting the TimeSpan in the tests and the UntilAsync method to a sufficiently long duration. Additionally, we can add an explanatory note in the user-facing method CreateAuthenticationTokenAsync to inform users about this issue in versions 3.2.X.

Or as I suggested [here](https://github.com/testcontainers/testcontainers-dotnet/pull/1103#discussion_r1576063779) set the default version to a LTS release that does not have the bug.

You can find the Issue in the Apache Pulsar Repo [here](https://github.com/apache/pulsar/issues/22811)

Reply:
> The mentioned bug affects all time units, not just seconds

Oh, that's unfortunate.

> Let’s collaborate to find the best path forward! 😊

I propse the following:

```csharp
int secondsToMilliseconds;

if (_configuration.Image.Tag.StartsWith("3.2"))
{
    Logger.LogWarning("The 'apachepulsar/pulsar:3.2.?' image contains a regression. The expiry time is converted to the wrong unit of time: https://github.com/apache/pulsar/issues/22811.");
    secondsToMilliseconds = 1000;
}
else
{
    secondsToMilliseconds = 1;
}

var command = new[]
{
    "bin/pulsar",
    "tokens",
    "create",
    "--secret-key",
    PulsarBuilder.SecretKeyFilePath,
    "--subject",
    PulsarBuilder.Username,
    "--expiry-time",
    $"{secondsToMilliseconds * expire.TotalSeconds}s",
};
```

WDYT?

Reply:
I think that works, I just checked and latest does indeed point to 3.2.3. So it should be good for now.
We can remove latest check, when the fix is released.
I think that this should unblock the test containers module?

Before you merge this PR in, will you make a GitHub do a git squash?

Reply:
> I think that this should unblock the test containers module?

👍 

> Before you merge this PR in, will you make a GitHub do a git squash?

Yes, I will take care of that.